pkg/csi/csi.go

```diff
@@ -0,0 +1,153 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi
+
+import (
+	"context"
+
+	"github.com/red-hat-storage/ocs-client-operator/pkg/templates"
+
+	"github.com/go-logr/logr"
+	appsv1 "k8s.io/api/apps/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+var (
+	// fetch the operator deployment and store it.
+	OperatorDeployment = &appsv1.Deployment{}
+)
+
+//+kubebuilder:rbac:groups="apps",resources=deployments,verbs=get;create;update;delete
+//+kubebuilder:rbac:groups="apps",resources=deployments/finalizers,verbs=update
+//+kubebuilder:rbac:groups="apps",resources=daemonsets,verbs=get;create;update;delete
+//+kubebuilder:rbac:groups="apps",resources=daemonsets/finalizers,verbs=update
+//+kubebuilder:rbac:groups="storage.k8s.io",resources=csidrivers,verbs=get;create;update;delete
+//+kubebuilder:rbac:groups="",resources=configmaps,verbs=create;update;delete
+//+kubebuilder:rbac:groups=security.openshift.io,resources=securitycontextconstraints,verbs=get;watch;create;patch;update
+//+kubebuilder:rbac:groups="",resources=serviceaccounts,verbs=get;create;update;delete
+//+kubebuilder:rbac:groups="",resources=serviceaccounts/finalizers,verbs=update
+//+kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=clusterroles,verbs=bind;escalate;get;create;update;delete
+//+kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=clusterroles/finalizers,verbs=update
+//+kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=clusterrolebindings,verbs=bind;escalate;get;create;update;delete
+//+kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=clusterrolebindings/finalizers,verbs=update
+//+kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=roles,verbs=bind;escalate;get;create;update;delete
+//+kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=roles/finalizers,verbs=update
+//+kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=rolebindings,verbs=bind;escalate;get;create;update;delete
+//+kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=rolebindings/finalizers,verbs=update
+
+// Providing escalate permissions on rolebindings and clusterrolebindings
+// to the operator is required to create the appropriate CSI driver roles.
+// More info:
+// https://kubernetes.io/docs/reference/access-authn-authz/rbac/#privilege-escalation-prevention-and-bootstrapping
+func EnsureCSI(ctx context.Context, client client.Client, log logr.Logger) error {
+	err := ensureRBAC(ctx, client, log)
+	if err != nil {
+		log.Error(err, "unable to create/update RBAC")
+		return err
+	}
+
+	// createEncryptionConfigMap creates the monitor configmap for the csi
+	// driver but never updates it. This is because the monitor configirations
+	// are added to the configmap when user creates storageclassclaims.
```

Comment:
```suggestion
	// createMonConfigMap creates the monitor configmap for the csi
	// driver but never updates it. This is because the monitor configurations
	// are added to the configmap when user creates storageclassclaims.
```